### DISCUSSION TITLE
Allow a single keyboard shortcut to trigger more actions

### DISCUSSION THREAD

**Discussion by axd1967** [Stellarium/stellarium#2751] [Ideas]
Currently, every keyboard shortcut can be linked to one action only.

The increasing amount of actions makes it harder to assign meaningful keyboard shortcuts.

This constraint could be relaxed. A shortcut could be assigned to several actions; a warning could be displayed during assignment (rather than when loading the shortcuts) rather than an error.

For example, I would like to assign a single shortcut to trigger the display of equinox as well as solstice points; another single shortcut to trigger all markings related to galactic marks (poles, equator, (anti)center).

This fits in a general philosophy to give a maximum of freedom to the user, rather than impose artificial (and sometimes arbitrary) limitations.

(Side note: concerning shortcuts, the secondary shortcuts should be left for exclusive use by the user.)

**Comment by alex-w** [comment]
Please write scripts with grouping the actions and assign shortcuts for these scripts. It's free!

> This fits in a general philosophy to give a maximum of freedom to the user, rather than impose artificial (and sometimes arbitrary) limitations.

You already have all needed tools for freely assign shortcuts for many groups of actions - why you ignore them? And no, we do not use "a general philosophy to give a maximum of freedom to the user" and we never used that!

> (Side note: concerning shortcuts, the secondary shortcuts should be left for exclusive use by the user.)

Please do not set the artificial limitations for users!

**Reply by axd1967** [reply]
> Please write scripts with grouping the actions and assign shortcuts for these scripts. It's free!

because scripts are advanced matter. I often write scripts, but I see a need for users that are less fluent with coding.

> > (Side note: concerning shortcuts, the secondary shortcuts should be left for exclusive use by the user.)
> 
> Please do not set the artificial limitations for users!

I don't. I impose a limitation on the developers, and offer more freedom to users.

**Reply by alex-w** [reply]
> because scripts are advanced matter. I often write scripts, but I see a need for users that are less fluent with coding.

Adding chaos and many points for confusing advanced users (I do not speak about nightmare for regular users here) is not making planetarium better, this making it worse.
 
> I don't. I impose a limitation on the developers, and offer more freedom to users.

You do it! Because only one way to implementing your proposal is blocking editing (and removing) primary shortcuts. As result you stolen freedom from users and developers and increasing chaos.

**Reply by axd1967** [reply]
Hello @alex-w 

I never asked to _prevent or forbid_ users to use the _primary_ shortcuts. users should retain full access to primary _and_ secondary shortcuts. Please avoid using words such as "chaotic", "nightmare", "worse", "confusing"...

it is is the **developers** that should remove their definitions of the secondary shortcuts, so that users always have access to all secondary shortcuts:

- Find: F3 / **Ctrl+F**
- Quit: Ctrl+Q / **Ctrl+X**
- Deepsky: D / **N**

That's all! This can be implemented with a change of 3 lines of code.

**Reply by alex-w** [reply]
> I never asked to prevent or forbid users to use the primary shortcuts. users should retain full access to primary and secondary shortcuts. 

Please read again all of your text. 

> Please avoid using words such as "chaotic", "nightmare", "worse", "confusing"...

If you proposes adding chaos then I say that you proposing adding chaos. Adding chaos and many source of confusing users making any software worse, not better.

> it is is the developers that should remove their definitions of the secondary shortcuts, so that users always have access to all secondary shortcuts

All of them was added for backward compatibility with previous releases and for compatibility with standard shortcuts for `find` / `quit` commands, and users can add own primary and secondary shortcuts freely (or change all of them). 

> That's all! This can be implemented with a change of 3 lines of code.

Yes, we can do it, but why we should broke user experience and why you think that only you is right? We do not forbid to you change any of shortcuts, but you want to break shortcuts for all users. This is definitely bad proposal, because it's fully imaginated.

**Reply by gzotti** [reply]
You can redefine primary and secondary shortcuts. There is no need for developers to remove their default assignments.

**Comment by gzotti** [comment]
+1 for developing 2-4-line scripts that fits your needs. This avoids unnecessary confusion.

**Comment by axd1967** [comment]
Nevertheless the main issue here is to allow the same shortcut to be assigned to multiple actions.

There IS a way to implement this proposal, by catching the event/action and forwarding to a **set** of related actions, rather than associating the event to a single (direct) action.

The default/existing behaviour remains the same, so anyone restricting shortcuts to one-on-one will not be impacted. At the same time, advanced users gain access to an additional functionality.

But we will all agree that this is not trivial. And I understand perfectly that there might be no enthusiasm to implement it.

**Reply by alex-w** [reply]
> Nevertheless the main issue here is to allow the same shortcut to be assigned to multiple actions.

You already have tool to do it, why you do not use then?

**Reply by axd1967** [reply]
> You already have tool to do it, why you do not use then?

I use it all the time (but the current behaviour does not allow to assign the same shortcut to multiple actions).

The idea of this feature request is to allow the user to trigger SEVERAL actions with the SAME shortcut.
